assets/js/util/tracking/index.js

```diff
@@ -0,0 +1,194 @@
+/**
+ * Event tracking utilities.
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const DEFAULT_CONFIG = {
+	isFirstAdmin: false,
+	trackingEnabled: false,
+	trackingID: '',
+	referenceSiteURL: '',
+	userIDHash: '',
+};
+
+/**
+ * Tracking configuration.
+ */
+const config = { ...DEFAULT_CONFIG };
+
+/**
+ * The promise created when tracking is enabled.
+ * This reference is kept and returned for subsequent calls.
+ */
+let enableTrackingPromise;
+
+/**
+ * gtag script identifier.
+ */
+const SCRIPT_IDENTIFIER = 'data-googlesitekit-gtag';
+
+/**
+ * Data layer global used for internal/private Site Kit data.
+ */
+export const DATA_LAYER = '_googlesitekitDataLayer';
```

Comment:
That's really neat that we can use our own global and pass it - will help with interoperability and scoping our own data.